src/components/DocSearch/DocSearch.js

```diff
@@ -0,0 +1,110 @@
+/**
+ * Copyright (c) 2013-present, Facebook, Inc.
+ *
+ * @emails react-core
+ */
+
+import React, {Component} from 'react';
+import {colors, media} from 'theme';
+
+class DocSearch extends Component {
+  state = {
+    disabled: false,
+  };
+  componentDidMount() {
+    // Initialize Algolia search.
+    // TODO Is this expensive? Should it be deferred until a user is about to search?
+    // eslint-disable-next-line no-undef
+    if (window.docsearch) {
+      window.docsearch({
+        apiKey: '36221914cce388c46d0420343e0bb32e',
+        indexName: 'react',
+        inputSelector: '#algolia-doc-search',
+      });
+    } else {
+      this.setState({disabled: true});
+    }
+  }
+
+  render() {
+    const {disabled} = this.state;
```

Comment:
I think we should just render `null` if search is unavailable. That's less confusing than a disabled search input I think. What do you think?

Reply:
Oh, sorry, I though you meant to disable the input when you said "3. Render the component in a disabled state if so". 😥 😃

Reply:
No worries at all. I was originally thinking the same, but then I tried this out and thought it might be better just to not show the input at all 😁